lib/godmin/engine_wrapper.rb

```diff
@@ -2,32 +2,47 @@ module Godmin
   class EngineWrapper
     attr_reader :engine
 
-    def initialize(obj)
-      @engine = find_engine(obj)
+    def initialize(controller)
+      @engine = find_engine(controller)
     end
 
-    def root
-      engine.root
+    def namespace
+      @namespace ||= engine.railtie_namespace
     end
 
-    def namespace
-      engine.railtie_namespace.to_s.underscore
+    def namespaced?
+      @namespaced ||= namespace.present?
+    end
+
+    def namespaced_path
```

Comment:
What do you think about renaming this? It may be confusing to return an array from a `_path` method. I'd expect a `Path` object or a string like "foo/bar/baz".